frontend/packages/topology/src/utils/application-utils.ts

```diff
@@ -179,6 +187,9 @@ export const cleanUpWorkload = async (
   const reqs = [];
   const buildConfigs = await k8sList(BuildConfigModel, { ns: resource.metadata.namespace });
   const builds = await k8sList(BuildModel, { ns: resource.metadata.namespace });
+  const channelModels = await fetchChannelsCrd();
+  const { group, kind } = getGroupVersionKindForResource(resource);
+  const resourceModel = modelFor(resource.kind) ?? modelForGroupKind(group, kind);
```

Comment:
can we just use `referenceFor`

```suggestion
  const resourceModel = modelFor(referenceFor(resource));
```